### PR TITLE
EntityStatus.Added using with additional parameter added

### PR DESCRIPTION
**Pull request by Calabonga**

### CODE REVIEW

File: src/Microsoft.EntityFrameworkCore.AutoHistory/Extensions/DbContextExtensions.cs [hunk 1]
```diff
@@ -20,17 +21,28 @@ public static class DbContextExtensions
         /// Ensures the automatic history.
         /// </summary>
         /// <param name="context">The context.</param>
-        public static void EnsureAutoHistory(this DbContext context)
+        /// <param name="useAddState">EntityStatus.Added will user too</param>
+        public static void EnsureAutoHistory(this DbContext context, bool useAddState = false)
```
Comment (xyting) [inline src/Microsoft.EntityFrameworkCore.AutoHistory/Extensions/DbContextExtensions.cs]: useAddState maybe rename to includeAddedState will be better?

Comment (Calabonga) [reply src/Microsoft.EntityFrameworkCore.AutoHistory/Extensions/DbContextExtensions.cs]: Ok. This is good too. Thanks

Comment (qsdfplkj) [inline src/Microsoft.EntityFrameworkCore.AutoHistory/Extensions/DbContextExtensions.cs:25]: ``
```suggestion
        public static void EnsureAutoHistory(this DbContext context, Func<EntityState, bool> predicate)
```

File: src/Microsoft.EntityFrameworkCore.AutoHistory/Extensions/DbContextExtensions.cs [hunk 2]
```diff
@@ -20,17 +21,28 @@ public static class DbContextExtensions
...
         }
 
-        public static void EnsureAutoHistory<TAutoHistory>(this DbContext context, Func<TAutoHistory> createHistoryFactory)
+        public static void EnsureAutoHistory<TAutoHistory>(this DbContext context, Func<TAutoHistory> createHistoryFactory, bool useAddState = false)
             where TAutoHistory : AutoHistory
         {
             // Must ToArray() here for excluding the AutoHistory model.
             // Currently, only support Modified and Deleted entity.
-            var entries = context.ChangeTracker.Entries().Where(e => e.State == EntityState.Modified || e.State == EntityState.Deleted).ToArray();
+            var allEntries = context.ChangeTracker.Entries();
```
Comment (qsdfplkj) [inline src/Microsoft.EntityFrameworkCore.AutoHistory/Extensions/DbContextExtensions.cs:35]: Maybe the user needs only deleted items to be logged.
```suggestion
            var allEntries = context.ChangeTracker.Entries().Where(e => e.State !=Detached && e.State != Unchanged && predicate(e.State));
```

**Pull request closed without merging**